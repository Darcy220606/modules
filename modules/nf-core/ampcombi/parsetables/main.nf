process AMPCOMBI_PARSETABLES {
    tag "$meta.id"
    label 'process_medium'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/ampcombi:0.2.1--pyhdfd78af_0':
        'biocontainers/ampcombi:0.2.1--pyhdfd78af_0' }"

    input:
    tuple val(meta), path(amp_input)
    path( faa_input )
    path( gbk_input )
    path( opt_amp_db )

    output:
    tuple val(meta), path("${meta.id}*")                        , emit: sample_dir
    tuple val(meta), path("${meta.id}/*diamond_matches.txt")    , emit: txt
    tuple val(meta), path("${meta.id}/*ampcombi.tsv")           , emit: tsv
    tuple val(meta), path("${meta.id}/*amp.faa")                , emit: faa
    tuple val(meta), path("${meta.id}/*_ampcombi.log")          , optional:true, emit: sample_log
    tuple val(meta), path("Ampcombi_parse_tables.log")          , optional:true, emit: full_log
    tuple val(meta), path("*/amp_ref_database")                 , optional:true, emit: results_db
    tuple val(meta), path("*/amp_ref_database/*.dmnd")          , optional:true, emit: results_db_dmnd
    tuple val(meta), path("*/amp_ref_database/*.clean.fasta")   , optional:true, emit: results_db_fasta
    tuple val(meta), path("*/amp_ref_database/*.tsv")           , optional:true, emit: results_db_tsv
    path "versions.yml"                                         , emit: versions


    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def db = opt_amp_db? "--amp_database $opt_amp_db": ""
    def faa = faa_input.isDirectory() ? "--faa ${faa_input}/" : "--faa ${faa_input}"
    def gbk = gbk_input.isDirectory() ? "--gbk ${gbk_input}/" : "--gbk ${gbk_input}"
    """
    ampcombi parse_tables \\
        $args \\
        --path_list '${amp_input.collect{"$it"}.join("' '")}' \\
        --sample_list ${prefix} \\
        --aminoacid_length 2000 \\
        --db_evalue 2000 \\
        --amp_cutoff 0 \\
        --ampir_file '.tsv' \\
        --amplify_file '.tsv' \\
        --macrel_file '.prediction' \\
        --neubi_file '.fasta' \\
        --hmmsearch_file 'candidates.txt' \\
        --ampgram_file '.tsv' \\
        --amptransformer_file '.txt' \\
        --threads $task.cpus \\
        ${db} \\
        ${faa} \\
        ${gbk}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        ampcombi: \$(ampcombi parse_tables --version | sed 's/ampcombi//')
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def db = opt_amp_db? "--amp_database $opt_amp_db": ""
    """
    mkdir -p ${prefix}
    touch ${prefix.id}/*diamond_matches.txt
    touch ${prefix}/*ampcombi.tsv
    touch ${prefix}/*amp.faa
    touch ${prefix}/*_ampcombi.log
    touch Ampcombi_parse_tables.log

    mkdir -p amp_ref_database
    touch amp_ref_database/*.dmnd
    touch amp_ref_database/*.clean.fasta
    touch amp_ref_database/*.tsv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        ampcombi: \$(ampcombi --version | sed 's/ampcombi//')
    END_VERSIONS
    """
}
