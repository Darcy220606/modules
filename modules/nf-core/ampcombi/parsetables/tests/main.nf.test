nextflow_process {

    name "Test Process AMPCOMBI_PARSETABLES"
    script "../main.nf"
    process "AMPCOMBI_PARSETABLES"
    tag "modules"
    tag "modules_nfcore"
    tag "antimicrobial peptides"
    tag "gunzip"
    tag "ampir"
    tag "amplify"
    tag "amplify_predict"
    tag "macrel_contigs"
    tag "ampcombi/parsetables"

    config "./nextflow.config"
    //setup {
        //run("PRODIGAL") {
        //    script "../../../prodigal/main.nf"
        //    process {
        //        """
        //        input[0] = [
        //                    [ id:'sample_test', single_end:false ], // meta map
        //                    file(params.test_data['sarscov2']['illumina']['contigs_fasta'], checkIfExists: true)
        //                   ]
        //        input[1] =  'gbk'
        //        """
        //    }
        //}
        //run("GUNZIP", alias:"GUNZIP_GBK") {
        //    script "../../../gunzip/main.nf"
        //    process {
        //        """
        //        input[0] =  PRODIGAL.out.gene_annotations
        //
        //        """
        //    }
        //}
        //run("GUNZIP", alias:"GUNZIP_FAA") {
        //    script "../../../gunzip/main.nf"
        //    process {
        //        """
        //        input[0] =  PRODIGAL.out.amino_acid_fasta
        //
        //        """
        //    }
        //}
        //setup {
        //    run("PYRODIGAL") {
        //        script "../../../pyrodigal/main.nf"
        //        process {
        //            """
        //            input[0] =  Channel.fromList([
        //                tuple([ id:'sample_1', single_end:false ], // meta map
        //                file(params.test_data['sarscov2']['illumina']['contigs_fasta'], checkIfExists: true))
        //            ])
        //            input[1] = 'gbk'
        //            """
        //        }
        //    }
        //    run("GUNZIP", alias: "GUNZIP_FAA") {
        //        script "../../../gunzip/main.nf"
        //        process {
        //            """
        //            input[0] =  PYRODIGAL.out.faa
//
        //            """
        //        }
        //    }
        //    //run("GUNZIP", alias: "GUNZIP_GBK") {
        //    //    script "../../../gunzip/main.nf"
        //    //    process {
        //    //        """
        //    //        input[0] =  PYRODIGAL.out.annotations
////
        //    //        """
        //    //    }
        //    //}
        //    run("AMPLIFY_PREDICT") {
        //        script "../../../amplify/predict/main.nf"
        //        process {
        //            """
        //            input[0] =  GUNZIP_FAA.out.gunzip
        //            input[1] = []
        //            """
        //        }
        //    }
        //run("AMPIR") {
        //    script "../../../ampir/main.nf"
        //    process {
        //        """
        //        input[0] = GUNZIP_FAA.out.gunzip
        //        input[1] = "precursor" // model
        //        input[2] = 10          // min_length
        //        input[3] = "0.5"       // min_probability
        //        """
        //    }
        //}
    //}
    setup {
        run("GUNZIP", alias:"GUNZIP_INPUT_GENOME") {
            script "../../../gunzip/main.nf"
                process {
                    """
                    input[0] =  Channel.fromList([
                        tuple([ id:'sample_1', single_end:false ], // meta map
                                file(params.test_data['bacteroides_fragilis']['illumina']['test1_contigs_fa_gz'], checkIfExists: true))
                    ])
                    """
            }
        }
        run("PYRODIGAL") {
            script "../../../pyrodigal/main.nf"
            process {
                """
                input[0] = GUNZIP_INPUT_GENOME.out.gunzip
                input[1] = 'gbk'
                """
            }
        }
        run("GUNZIP", alias:"GUNZIP_GBK") {
            script "../../../gunzip/main.nf"
            process {
                """
                input[0] =  PYRODIGAL.out.annotations

                """
            }
        }
        run("GUNZIP", alias:"GUNZIP_FAA") {
            script "../../../gunzip/main.nf"
            process {
                """
                input[0] =  PYRODIGAL.out.faa

                """
            }
        }
        //run("AMPLIFY_PREDICT") {
        //    script "../../../amplify/predict/main.nf"
        //    process {
        //        """
        //        input[0] = GUNZIP_FAA.out.gunzip
        //        input[1] = []
        //        """
        //    }
        //}
        run("MACREL_CONTIGS") {
            script "../../../macrel/contigs/main.nf"
            process {
                """
                input[0] =  [
                             [id:'sample_1', single_end:false ], // meta map
                             file( GUNZIP_INPUT_GENOME.out.gunzip.map{ it[1] } ) //file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)
                            ]
                """
            }
        }
        run("GUNZIP", alias:"GUNZIP_MACREL") {
            script "../../../gunzip/main.nf"
            process {
                """
                input[0] =  MACREL_CONTIGS.out.amp_prediction
                """
            }
        }

    }


    test("ampcombi_parsetables - sarscov2") {

        when {
            process {
                """
                //ch_amplify = AMPLIFY_PREDICT.out.tsv
                ch_amplify = GUNZIP_MACREL.out.gunzip
                ch_ampir = GUNZIP_MACREL.out.gunzip

                ch_amplify
                    .mix( ch_ampir )
                    .map{
                        [ it[1] ]
                    }
                    .collect()
                    .set { ch_input_for_ampcombi }

                input[0] = [
                    [ id:'sample_1'],
                    [ ch_input_for_ampcombi ]
                ]
                input[1] = GUNZIP_FAA.out.gunzip.map{ it[1] }
                input[2] = GUNZIP_GBK.out.gunzip.map{ it[1] }
                input[3] = []
                """
            }
        }

        then {
            assertAll(
                { assert process.success }
                //{ assert snapshot(process.out).match() }
            )
        }
    }
}

//    test("sarscov2") {
//
//        options "-stub"
//
//        when {
//            process {
//                """
//                // TODO nf-core: define inputs of the process here. Example:
//
//                input[0] = [
//                    [ id:'test', single_end:false ], // meta map
//                    file(params.test_data['sarscov2']['illumina']['test_paired_end_bam'], checkIfExists: true)
//                    ]
//                """
//            }
//        }
//
//        then {
//            assertAll(
//                { assert process.success },
//                { assert snapshot(process.out).match() }
//            )
//        }
//
//    }
//}
