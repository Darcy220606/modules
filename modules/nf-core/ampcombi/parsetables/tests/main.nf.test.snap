{
    "ampcombi_parsetables - bacteroides_fragilis - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "sample_1",
                            "single_end": false
                        },
                        [
                            [
                                
                            ],
                            "sample_1_amp.faa:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "sample_1_ampcombi.log:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "sample_1_ampcombi.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "sample_1_diamond_matches.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "sample_1",
                            "single_end": false
                        },
                        [
                            
                        ]
                    ]
                ],
                "10": [
                    [
                        {
                            "id": "sample_1",
                            "single_end": false
                        },
                        "*.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "11": [
                    "versions.yml:md5,0308fdaccfdf7b96225fe8c67032b261"
                ],
                "2": [
                    [
                        {
                            "id": "sample_1",
                            "single_end": false
                        },
                        "sample_1_diamond_matches.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "sample_1",
                            "single_end": false
                        },
                        "sample_1_ampcombi.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "4": [
                    [
                        {
                            "id": "sample_1",
                            "single_end": false
                        },
                        "sample_1_amp.faa:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "5": [
                    [
                        {
                            "id": "sample_1",
                            "single_end": false
                        },
                        "sample_1_ampcombi.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "6": [
                    [
                        {
                            "id": "sample_1",
                            "single_end": false
                        },
                        "Ampcombi_parse_tables.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "7": [
                    [
                        {
                            "id": "sample_1",
                            "single_end": false
                        },
                        [
                            "*.clean.fasta:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "*.dmnd:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "*.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "8": [
                    [
                        {
                            "id": "sample_1",
                            "single_end": false
                        },
                        "*.dmnd:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "9": [
                    [
                        {
                            "id": "sample_1",
                            "single_end": false
                        },
                        "*.clean.fasta:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "contig_gbks": [
                    [
                        {
                            "id": "sample_1",
                            "single_end": false
                        },
                        [
                            
                        ]
                    ]
                ],
                "faa": [
                    [
                        {
                            "id": "sample_1",
                            "single_end": false
                        },
                        "sample_1_amp.faa:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "full_log": [
                    [
                        {
                            "id": "sample_1",
                            "single_end": false
                        },
                        "Ampcombi_parse_tables.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "results_db": [
                    [
                        {
                            "id": "sample_1",
                            "single_end": false
                        },
                        [
                            "*.clean.fasta:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "*.dmnd:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "*.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "results_db_dmnd": [
                    [
                        {
                            "id": "sample_1",
                            "single_end": false
                        },
                        "*.dmnd:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "results_db_fasta": [
                    [
                        {
                            "id": "sample_1",
                            "single_end": false
                        },
                        "*.clean.fasta:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "results_db_tsv": [
                    [
                        {
                            "id": "sample_1",
                            "single_end": false
                        },
                        "*.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "sample_dir": [
                    [
                        {
                            "id": "sample_1",
                            "single_end": false
                        },
                        [
                            [
                                
                            ],
                            "sample_1_amp.faa:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "sample_1_ampcombi.log:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "sample_1_ampcombi.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "sample_1_diamond_matches.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "sample_log": [
                    [
                        {
                            "id": "sample_1",
                            "single_end": false
                        },
                        "sample_1_ampcombi.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "tsv": [
                    [
                        {
                            "id": "sample_1",
                            "single_end": false
                        },
                        "sample_1_ampcombi.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "txt": [
                    [
                        {
                            "id": "sample_1",
                            "single_end": false
                        },
                        "sample_1_diamond_matches.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,0308fdaccfdf7b96225fe8c67032b261"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-27T09:29:00.919910492"
    },
    "ampcombi_parsetables - bacteroides_fragilis": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "sample_1",
                            "single_end": false
                        },
                        [
                            [
                                
                            ],
                            "sample_1_amp.faa:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "sample_1_ampcombi.log:md5,db6855ff5bca822fcdb4f110ea7ba7e1",
                            "sample_1_ampcombi.tsv:md5,b75e2eb2fd94eda39ab3c2b9032c2808",
                            "sample_1_diamond_matches.txt:md5,3dba5c186c9d3a87a8e9aaa5db2af521"
                        ]
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "sample_1",
                            "single_end": false
                        },
                        [
                            
                        ]
                    ]
                ],
                "10": [
                    [
                        {
                            "id": "sample_1",
                            "single_end": false
                        },
                        "general_amps_2024_03_27.tsv:md5,77310475ad800e538cc06185ff145040"
                    ]
                ],
                "11": [
                    "versions.yml:md5,0308fdaccfdf7b96225fe8c67032b261"
                ],
                "2": [
                    [
                        {
                            "id": "sample_1",
                            "single_end": false
                        },
                        "sample_1_diamond_matches.txt:md5,3dba5c186c9d3a87a8e9aaa5db2af521"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "sample_1",
                            "single_end": false
                        },
                        "sample_1_ampcombi.tsv:md5,b75e2eb2fd94eda39ab3c2b9032c2808"
                    ]
                ],
                "4": [
                    [
                        {
                            "id": "sample_1",
                            "single_end": false
                        },
                        "sample_1_amp.faa:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "5": [
                    [
                        {
                            "id": "sample_1",
                            "single_end": false
                        },
                        "sample_1_ampcombi.log:md5,db6855ff5bca822fcdb4f110ea7ba7e1"
                    ]
                ],
                "6": [
                    [
                        {
                            "id": "sample_1",
                            "single_end": false
                        },
                        "Ampcombi_parse_tables.log:md5,0f83d19739214b8a2069b2218c696ade"
                    ]
                ],
                "7": [
                    [
                        {
                            "id": "sample_1",
                            "single_end": false
                        },
                        [
                            "amp_ref.dmnd:md5,f20d8d0c073bcc5857962f4bf62ace9d",
                            "general_amps_2024_03_27.tsv:md5,77310475ad800e538cc06185ff145040",
                            "general_amps_2024_03_27_clean.fasta:md5,05c063716bbb55c37ae33bda4e0e6732"
                        ]
                    ]
                ],
                "8": [
                    [
                        {
                            "id": "sample_1",
                            "single_end": false
                        },
                        "amp_ref.dmnd:md5,f20d8d0c073bcc5857962f4bf62ace9d"
                    ]
                ],
                "9": [
                    
                ],
                "contig_gbks": [
                    [
                        {
                            "id": "sample_1",
                            "single_end": false
                        },
                        [
                            
                        ]
                    ]
                ],
                "faa": [
                    [
                        {
                            "id": "sample_1",
                            "single_end": false
                        },
                        "sample_1_amp.faa:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "full_log": [
                    [
                        {
                            "id": "sample_1",
                            "single_end": false
                        },
                        "Ampcombi_parse_tables.log:md5,0f83d19739214b8a2069b2218c696ade"
                    ]
                ],
                "results_db": [
                    [
                        {
                            "id": "sample_1",
                            "single_end": false
                        },
                        [
                            "amp_ref.dmnd:md5,f20d8d0c073bcc5857962f4bf62ace9d",
                            "general_amps_2024_03_27.tsv:md5,77310475ad800e538cc06185ff145040",
                            "general_amps_2024_03_27_clean.fasta:md5,05c063716bbb55c37ae33bda4e0e6732"
                        ]
                    ]
                ],
                "results_db_dmnd": [
                    [
                        {
                            "id": "sample_1",
                            "single_end": false
                        },
                        "amp_ref.dmnd:md5,f20d8d0c073bcc5857962f4bf62ace9d"
                    ]
                ],
                "results_db_fasta": [
                    
                ],
                "results_db_tsv": [
                    [
                        {
                            "id": "sample_1",
                            "single_end": false
                        },
                        "general_amps_2024_03_27.tsv:md5,77310475ad800e538cc06185ff145040"
                    ]
                ],
                "sample_dir": [
                    [
                        {
                            "id": "sample_1",
                            "single_end": false
                        },
                        [
                            [
                                
                            ],
                            "sample_1_amp.faa:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "sample_1_ampcombi.log:md5,db6855ff5bca822fcdb4f110ea7ba7e1",
                            "sample_1_ampcombi.tsv:md5,b75e2eb2fd94eda39ab3c2b9032c2808",
                            "sample_1_diamond_matches.txt:md5,3dba5c186c9d3a87a8e9aaa5db2af521"
                        ]
                    ]
                ],
                "sample_log": [
                    [
                        {
                            "id": "sample_1",
                            "single_end": false
                        },
                        "sample_1_ampcombi.log:md5,db6855ff5bca822fcdb4f110ea7ba7e1"
                    ]
                ],
                "tsv": [
                    [
                        {
                            "id": "sample_1",
                            "single_end": false
                        },
                        "sample_1_ampcombi.tsv:md5,b75e2eb2fd94eda39ab3c2b9032c2808"
                    ]
                ],
                "txt": [
                    [
                        {
                            "id": "sample_1",
                            "single_end": false
                        },
                        "sample_1_diamond_matches.txt:md5,3dba5c186c9d3a87a8e9aaa5db2af521"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,0308fdaccfdf7b96225fe8c67032b261"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-27T09:28:38.699495071"
    }
}